.github/workflows/container-image.yml

```diff
@@ -0,0 +1,93 @@
+name: Build Container Image
+
+concurrency:
+  group: ci-${{ github.run_id }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * 1' # every Monday at midnight
+  push:
+    branches:
+      - main
+      - v[0-9]+
+      - v[0-9]+.[0-9]+
+      - cryostat-v[0-9]+.[0-9]+
+
+env:
+  OPENSUSE_UNOFFICIAL_LIBCONTAINERS_KEY_URL: "https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04/Release.key"
+  OPENSUSE_UNOFFICIAL_LIBCONTAINERS_SOURCE_URL: "https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04"
+  CI_USER: redhat-java-monitoring+github_ci
+  CI_REGISTRY: quay.io
+  CI_NAMESPACE: redhat-java-monitoring
+
+jobs:
+  get-agent-version:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository_owner == 'cryostatio' }}
+    outputs: 
+      image-version: ${{ steps.get-agent-version.outputs.agent-version }}
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: ${{ github.repository_owner }}/cryostat-agent
+        ref: main
+        submodules: true
+        fetch-depth: 0
+    - id: get-agent-version
+      run: |
+        echo "agent-version=$(mvn -q -DforceStdout help:evaluate -Dexpression=project.version)" >> $GITHUB_OUTPUT
+    - run: mvn -B -U clean install
+    - uses: actions/cache@v4
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-build-${{ hashFiles('**/pom.xml') }}
+
+  build-and-push:
+    needs: [get-agent-version]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: ['17']
+    outputs:
+      quay-image: ${{ steps.quarkus-test-image.outputs.image }}
+    env:
+      agent-version: ${{ needs.get-agent-version.outputs.image-version }}
+    name: Build Java ${{ matrix.java }}
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+        fetch-depth: 0
+    - name: Install qemu
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y qemu-user-static
+    - uses: actions/setup-java@v4
+      with:
+        java-version: ${{ matrix.java }}
+        distribution: 'temurin'
+    - uses: actions/cache@v4
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-build-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
```

Comment:
I don't think this is going to always hit the key we want for the most recent cache. Probably want to checkout the -agent repo first since the earlier key involves a hash of its pom.xml

Reply:
oh is this trying to restore the cache from line 90? Either way, there is no pom.xml file in this repo, so the key should be replaced.

Reply:
There is no `pom.xml` in the root of the repo, but the `**` glob pattern will find the `quarkus-test-agent/pom.xml` I think. That's a pretty bad way to do the cache keying though, you are right about that.